dev/DynamicDependency/WinRTInprocModule.h

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if !defined(WINRTINPROCMODULE_H)
+#define WINRTINPROCMODULE_H
+
+#include <activationregistration.h>
+#include <activation.h>
+
+#include "MddWinRT.h"
+
+//TODO Components won't respect COM lifetime. workaround to get them in the COM list? See dev\UndockedRegFreeWinRT\catalog.cpp
+
+namespace MddCore
+{
+class WinRTInprocModule
+{
+public:
+    WinRTInprocModule() = default;
+
+    WinRTInprocModule(WinRTInprocModule&& other) :
+        m_path(std::move(other.m_path)),
+        m_dll(std::move(other.m_dll)),
+        m_dllGetActivationFactory(std::move(other.m_dllGetActivationFactory))
+    {
+        m_inprocServers.insert(std::make_move_iterator(other.m_inprocServers.begin()), std::make_move_iterator(other.m_inprocServers.end()));
+    }
+
+    ~WinRTInprocModule() = default;
+
+    MddCore::WinRT::ThreadingModel Find(const std::wstring& activatableClassId)
+    {
+        auto iterator = m_inprocServers.find(activatableClassId);
+        if (iterator != m_inprocServers.end())
+        {
+            return iterator->second;
+        }
+        return MddCore::WinRT::ThreadingModel::Unknown;
+    }
+
+    void* GetActivationFactory(
+        HSTRING className,
+        const std::wstring& activatableClassId,
+        REFIID iid)
+    {
+        Load();
+
+        IActivationFactory* ifactory{};
+        THROW_IF_FAILED(m_dllGetActivationFactory(className, &ifactory));
+
+        //TODO optimize for IActivationFactory? See GetActivationFactory() in dev\UndockedRegFreeWinRT\catalog.cpp
```

Comment:
[External]
TODO. Not sure what policy dictates here but I suspect this should be deleted.

Reply:
This is a reference to the comment in UndockedRegFreeWinRT

I'm following up with the URFW folks to discuss solutions. Will be addressed in a future PR

See [MSIX Dynamic Dependencies PR follow ups #403](https://github.com/microsoft/ProjectReunion/issues/403) for more details